src/otx/core/data/transform_libs/mmcv.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Helper to support MMCV data transform functions."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Callable
+
+import numpy as np
+from mmcv.transforms import LoadImageFromFile as MMCVLoadImageFromFile
+from mmcv.transforms.builder import TRANSFORMS
+
+from otx.core.data.entity.base import OTXDataEntity
+from otx.core.utils.config import convert_conf_to_mmconfig_dict
+
+if TYPE_CHECKING:
+    from mmengine.registry import Registry
+
+    from otx.core.config.data import SubsetConfig
+
+
+@TRANSFORMS.register_module(force=True)
+class LoadImageFromFile(MMCVLoadImageFromFile):
+    """Class to override MMCV LoadImageFromFile."""
+
+    def transform(self, entity: OTXDataEntity) -> dict | None:
+        """Transform OTXDataEntity to MMCV data entity format."""
+        img = entity.image
+
+        if self.to_float32:
+            img = img.astype(np.float32)
+
+        results = {}
+        results["img"] = img
+        results["img_shape"] = img.shape[:2]
+        results["ori_shape"] = img.shape[:2]
+
+        results["__otx__"] = entity
+
+        return results
+
+
+class MMCVTransformLib:
+    """Helper to support MMCV transforms in OTX."""
+
+    @classmethod
+    def get_builder(cls) -> Registry:
+        """Transform builder obtained from MMCV."""
+        return TRANSFORMS
+
+    @classmethod
+    def _check_mandatory_transforms(
+        cls,
+        transforms: list[Callable],
+        mandatory_transforms: set,
+    ) -> None:
+        for transform in transforms:
+            t_transform = type(transform)
+            mandatory_transforms.discard(t_transform)
+
+        if len(mandatory_transforms) != 0:
+            msg = f"{mandatory_transforms} should be included"
+            raise RuntimeError(msg)
+
+    @classmethod
+    def generate(cls, config: SubsetConfig) -> list[Callable]:
+        """Generate MMCV transforms from the configuration."""
+        transforms = [
```

Comment:
If other `MMXFrameworks` have exact same `LoadImageFromFile` transform, this register line (https://github.com/openvinotoolkit/training_extensions/pull/2670/files#diff-80ae88fef69798189bd2d5b2ede32f41ff2f2922fe4d5dac96cd6ca676de2981R23) overrides the LoadImageFromFile located at the parent registry. So, we should consider the same name case. So, want to suggest the lines below,
```python3
# If builder already has same name transform (i.e.MMPretrain.LoadImageFromFile)
# builder and builder.parent have different class since we already overridded 
# the transform at the src/otx/core/data/transfor_libs/mmcv.py, line#23
if builder.get(converted_cfg.type) != builder.parent.get(converted_cfg.type):
    build_transform = builder.parent.build(converted_cfg)
# If the transform is not duplicated, builder and builder.parent have same class.
else:
    build_transform = builder.build(converted_cfg)
```

In the MMPretrain, there was already `LoadImageFromFile`, So, `Registry.register_module(force=True)` overrides the transform located at the parent registry, however, it should change the transform located at the self class.

---

In this case, builder.parent's transform registry successfully updated as what we wanted,
<img width="397" alt="image" src="https://github.com/openvinotoolkit/training_extensions/assets/80735344/255fc7e0-c207-4a8c-8189-ef4fe9fe04ed">

However, mmpretrain's transform registry wasn't updated.
<img width="459" alt="image" src="https://github.com/openvinotoolkit/training_extensions/assets/80735344/9796152d-3c78-41fc-81e4-85e59015801e">

So, it raises an error due to the `_check_mandatory_transforms()`
<img width="446" alt="image" src="https://github.com/openvinotoolkit/training_extensions/assets/80735344/300a43e5-8f39-40a3-bdc1-d27131d93984">

Reply:
Sorry I don't get you point. From your comment, I can see that `mmpretrain` uses `LoadImageFromFile` from `mmcv` as same as `mmdetection`.
![](https://private-user-images.githubusercontent.com/80735344/285817789-9796152d-3c78-41fc-81e4-85e59015801e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDEwODA2NjAsIm5iZiI6MTcwMTA4MDM2MCwicGF0aCI6Ii84MDczNTM0NC8yODU4MTc3ODktOTc5NjE1MmQtM2M3OC00MWZjLTgxZTQtODVlNTkwMTU4MDFlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzExMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMTI3VDEwMTkyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRlMzlkNzg1MjNhMDMzYTQzNjE3MTAyN2U4NGE2NGFlM2Q5MmZhMmU3YjlmNDg3ODk2ZjQ1YjZkYWZjY2I2MmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.10VZecD6hxt54_XOLPPyfPk_hcJ5S8HwDfnrFNc73Jk)

`mmdetection` uses `LoadImageFromFile` from `mmcv` and it is registered in this object:
```python
from mmcv.transforms.builder import TRANSFORMS
```

Do you mean that `mmpretrain`'s `LoadImageFromFile` is registered in its own registry? not registered that object `mmcv.transforms.builder.TRANSFORMS`?

Reply:
I discussed with @vinnamkim , and we decided that leave this code AS-IS, for the MMPretrain case, I'll make workaround at my branch and then consider the solution at the next phase. So, I resolved this.

Reply:
Is this working?

```python
# This is `src/otx/core/data/transform_libs/mmpretrain.py` file

from mmpretrain.registry import TRANSFORMS
from .mmcv import LoadImageFromFile

TRANSFORMS.register_module(LoadImageFromFile)

...
...
...
```